### ISSUE TITLE
Very noticeable remote cursor delay 

### ISSUE THREAD

**Issue by ShaneTRS** [TigerVNC/tigervnc#1563]
**Describe the bug**
When connecting to a remote VNC server via `tigervnc`'s `vncviewer`, the remote cursor isn't updated in real-time. It seems like there's a deliberate delay or polling rate that's incredibly low,

**To Reproduce**
1. Connect to a VNC server
2. Move the mouse connected to the remote server (or use something else, like an automation tool)

**Expected behavior**
When the person on the other end moves the mouse, I expect the cursor position to update every frame. Just like it does in applications with a custom cursor.

**Client (please complete the following information):**
 - OS: Arch Linux x86_64
 - VNC client: TigerVNC
 - VNC client version: v1.12.0
 - Client downloaded from: [aur/tigervnc-git](https://aur.archlinux.org/packages/tigervnc-git)

**Server (please complete the following information):**
 - OS: Arch Linux x86_64
 - VNC server: TigerVNC
 - VNC server version: v1.12.0
 - Server downloaded from: [community/tigervnc](https://archlinux.org/packages/community/x86_64/tigervnc/)
 - Server was started using: `x0vncserver`

**Additional context**
I use a wireless mouse as a workaround for #619
Position of the relevant code would be nice, but I'm having trouble finding it. Fixing it locally, if it's never changed up-stream, would be a proper solution for me.

**Comment by CendioOssman** [comment]
I did a test here and I'm afraid I'm not seeing any big delay in cursor movement.

There is indeed a polling rate (which is a bit dynamic), which means it won't be achieving any high frame rates. But it should still be updating a few times a second.

Is the issue you are seeing the fact that it isn't running a buttery smooth 60+ fps?

> I use a wireless mouse as a workaround for #619

I'm afraid I don't understand how using a wireless mouse works around that issue?

**Comment by ShaneTRS** [comment]
Yes, I think the polling rate is what I was referring to (though some applications are unaffected, such as Stardew Valley).
It is indeed a few times a second, which makes it fairly difficult to control.
The wireless mouse works because I'm within bluetooth range of the host, so I can have relative mouse movements (as it's actually connected to the host machine itself, not via the client). 
Would you happen to know where the polling rate is kept in source? You said it's dynamic, so I won't find a hardcoded figure, but the logic for it would help me too

**Comment by CendioOssman** [comment]
It's mostly in here: https://github.com/TigerVNC/tigervnc/blob/master/unix/x0vncserver/PollingScheduler.cxx

That polling was originally for updating the screen, and updating the mouse was just a side effect. These days, we use XDAMAGE, which means we no longer use polling for the screen. But that polling scheduler still drives when the mouse is updated:

https://github.com/TigerVNC/tigervnc/blob/4895f53aef52f878a1176f6e4f0f2992553d560c/unix/x0vncserver/x0vncserver.cxx#L429-L432

https://github.com/TigerVNC/tigervnc/blob/4895f53aef52f878a1176f6e4f0f2992553d560c/unix/x0vncserver/XDesktop.cxx#L228

**Comment by CendioOssman** [comment]
We could probably solve this efficiently by listening to XInput devices, instead of X11 core devices. That API allows us to better monitor whenever the mouse moves, regardless of which application happens to be under it.
